getting_started/scripting/gdscript/gdscript_basics.rst

```diff
@@ -1463,49 +1463,88 @@ freed.
+   # A signal that forwards two arguments
+   signal health_changed(old_value, new_value)
 
-These signals may be connected to methods in the same manner as you connect
-built-in signals of nodes such as :ref:`class_Button` or :ref:`class_RigidBody`.
+.. note:: 
 
-Here's an example that creates a custom signal in one script and connects
-the custom signal to a method in a separate script, using the
+   Signals are a `Callback
+          <https://www.wikiwand.com/en/Callback_(computer_programming)>`
+          mechanism. They also fill the role of Observers, a common programming
+          pattern. For more information, read the `Observer
+          tutorial<http://gameprogrammingpatterns.com/observer.html>`_ in the
+          Game Programming Patterns ebook.
+
+You can connect these signals to methods the same way you connect built-in
+signals of nodes like :ref:`class_Button` or :ref:`class_RigidBody`.
+
+Here's an example of when signals can be useful. 
+
+Let's say we have a game where the playable character can take damage from
+enemies. We want a life bar on screen to react to health changes with an
+animation, but we want to keep the User Interface separate from the player in
+our scene tree.
+
+In our ``Character.gd`` script, we define a ``health_changed`` signal and emit
+it with :ref:`Object.emit_signal() <class_Object_method_emit_signal>`, and from
+a ``Game`` node higher up our scene tree, we connect it to the ``Lifebar`` using the
 :ref:`Object.connect() <class_Object_method_connect>` method:
 
 ::
 
-   # your_notifier.gd
+    # Character.gd
 
-   signal data_found
+    ...
+    signal health_changed(old_value, new_value)
 
-   var your_data = 42
+    func take_damage(amount):
+        var old_health = health
+        health -= amount
+
+        # We emit the health_changed signal every time the
+        # character took damage
+        emit_signal('health_changed', old_health, health)
+    ...
 
 ::
 
-   # your_handler.gd
+    # Lifebar.gd 
 
-   func your_handler():
-      print("Your handler was called!")
+    # Here, we define a function to use as a callback when the
+    # character's health_changed signal was emitted
+
+    ...
+    func _on_Character_health_changed(old_value, new_value):
+        if old_value > new_value:
+            progress_bar.color = Color.red
+        else:
+            progress_bar.color = Color.green
+        progress_bar.animate(old_value, new_value)
+    ...
+
+
+In the ``Game`` node, we get both the ``Character`` and ``Lifebar`` nodes, then connect the character, that emits the signal, to the receiver, the ``Lifebar`` node in this case.
 
 ::
 
-   # your_game.gd
+   # Game.gd
+
 
    func _ready():
-      var notifier = your_notifier.new()
-      var handler = your_handler.new()
+      var character_node = get_node('Character')
+      var lifebar_node = get_node('UserInterface/Lifebar')
+
+      character_node.connect("health_changed", lifebar_node, "_on_Character_health_changed")
```

Comment:
Maybe also include an example with connecting through the Node tab, instead of using the `connect` function?

Reply:
This is the gdscript code/features reference, so I'm not sure it should be there.

Reply:
I mean i feel like this would better fit a tutorial dedicated to signals, as you'd then have to run the user through the node tab, the signals connection interface and its different buttons... plus talk about the advantages of one approach over the other.